File: 2363-merge-similar-items/2363-merge-similar-items.cpp
class Solution {
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2) {
     int arr[1001]={0};
        vector<vector<int>> result;
        
        for(int i=0; i<items1.size(); i++)
        {
            arr[items1[i][0]]=items1[i][1];
        }
        for(int i=0; i<items2.size(); i++)
        {
            arr[items2[i][0]]+=items2[i][1];
        }
        for(int i=0; i<1001; i++)
        {
            vector<int> v;
            if(arr[i]!=0)
            {
                v.push_back(i);
                v.push_back(arr[i]);
                result.push_back(v);
            }
        }
        return result;
    }
};

File: 2363-merge-similar-items/2363-merge-similar-items.java
class Solution {
    public List<List<Integer>> mergeSimilarItems(int[][] items1, int[][] items2) {
        List<List<Integer>> result = new ArrayList<>();
       
        int tempArr[] =new int[1001];
        for(int i=0; i<items1.length; i++)
        {
            int val = items1[i][0];
            int weight=items1[i][1];
            tempArr[val]=weight;
        }
        
        for(int i=0; i<items2.length;i++)
        {
            int val = items2[i][0];
            int weight=items2[i][1];
            tempArr[val]+=weight;
        }
        
        for(int i=0;i<tempArr.length;i++)
        {
            if(tempArr[i]>0)
            {
                List<Integer> ls= new ArrayList<>();
                ls.add(i);
                ls.add(tempArr[i]);
                result.add(ls);
            }
        }
      return result;
    }
}